### PR TITLE
Added cars that follow player 0

### PR DESCRIPTION
**Pull request by Pigglesworth**

### CODE REVIEW

File: src/AiSystem.cpp [hunk 1]
```diff
@@ -2,59 +2,149 @@
...
-                    "player_id", 3 - i,
-                    "key", SDL_CONTROLLER_AXIS_TRIGGERRIGHT,
-                    "value", rand() % 32768
-                )
-            );
+
+    for (int i = 3; i >= 4 - num_ai_; i--) {
+        if (i != whos_it_) {
+            path_to(i, cars_[whos_it_].car_transform_.position_);
+        } else if (counter_ == 0) {
```
Comment (enochtsang) [inline src/AiSystem.cpp]: shoudln't this just be else?

File: src/AiSystem.h [hunk 2]
```diff
@@ -9,6 +13,23 @@ class AiSystem : public EventSystem<AiSystem> {
...
 private:
     void handle_activate_ai(const Event& e);
+    void handle_object_transform(const Event& e);
+
+    void path_to(int car, const glm::vec3& point_);
+
     int num_ai_;
+
+    struct car_data {
+        struct car_transform_data {
```
Comment (enochtsang) [inline src/AiSystem.h]: Isn't this struct unnecessary?

Just `car_transform_data.rotation_` should suffice right?

File: src/AiSystem.cpp [hunk 3]
```diff
@@ -2,59 +2,149 @@
...
+
+    if (angle < -0.05) {
+        axis = -axis;
+
+        EventSystem::queue_event(
+            Event(
+                EventType::KEYPRESS_EVENT,
+                "player_id", car,
+                "key", SDL_CONTROLLER_AXIS_LEFTX,
+                "value", axis
```
Comment (enochtsang) [inline src/AiSystem.cpp]: Just passing `-axis` here is easier to follow.

File: src/AiSystem.cpp [hunk 4]
```diff
@@ -2,59 +2,149 @@
...
+            "player_id", car,
+            "key", SDL_CONTROLLER_AXIS_TRIGGERRIGHT,
+            "value", SHRT_MAX
+        )
+    );
+
+
+    std::cout << "car " << car << " angle " << angle << " axis " << axis << std::endl;
+    std::cout << "diff1 " << vec2_to.x << " " << vec2_to.y << std::endl;
+    std::cout << "diff2 " << vec2_facing.x << " " << vec2_facing.y << std::endl;
```
Comment (enochtsang) [inline src/AiSystem.cpp]: Can we comment this out before merging?

File: src/AiSystem.cpp (hunk 1 above)
Comment (Pigglesworth) [reply src/AiSystem.cpp]: If they're it, we only want to change their random controlling every 10 frames.

File: src/AiSystem.h (hunk 2 above)
Comment (Pigglesworth) [reply src/AiSystem.h]: We might want to add more to this struct in the future, like what type of power-up they have

**Pull request merged**